### PR TITLE
Add first-class auth login config support

### PR DESCRIPTION
**Pull request by benashz**
This fix provides first-class auth specific configuration for logging into Vault. Initial auth engines supported are userpass and aws.

This is an additive change, which means that the current auth_login functionality remains unchanged.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note

```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: internal/provider/auth_aws.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	AuthLoginCommon
+}
+
+func (l *AuthLoginAWS) Schema() *schema.Resource {
+	return SchemaLoginAWS
+}
+
+func (l *AuthLoginAWS) MountPath() string {
+	if l.mount == "" {
+		return consts.AuthMethodUserpass
```
Comment (vinay-gopalan) [inline internal/provider/auth_aws.go:117]: should this be `consts.AuthMethodAWS`?

Comment (benashz) [reply internal/provider/auth_aws.go:117]: Doh! Yes.

Comment (benashz) [reply internal/provider/auth_aws.go]: Fixed in 640f974bad593f178a026097b5acb997608283d9

**Review by vinay-gopalan — APPROVED** [review]
Looks good!

**Pull request merged**